website_sale_commitment_date/models/delivery.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2025 Binhex - Adasat Torres de León
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import calendar
+from datetime import date, timedelta
+
+from babel.dates import get_month_names
+
+from odoo import api, fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    allow_commitment_date = fields.Boolean(
+        string="Allow commitment date",
+        help="Allow to set a commitment date on the sale order from to the e-commerce",
+    )
+
+    min_commitment_days = fields.Float()
+    max_commitment_days = fields.Float()
+
+    exclude_weekday_ids = fields.Many2many(
+        comodel_name="delivery.carrier.weekday",
+        relation="delivery_carrier_weekday_rel",
+        string="Exclude Weekdays",
+    )
+
+    exclude_date_ids = fields.Many2many(
+        comodel_name="delivery.carrier.date",
+        relation="delivery_carrier_date_rel",
+        string="Exclude Dates",
+    )
+
+    def _get_calendar_context(self, start=False):
+        today = fields.Date.today()
+
+        if start:
+            year, month = start.year, start.month
+        else:
+            year, month = today.year, today.month
+            start = date(year, month, 1)
```

Comment:
```suggestion
            start = fields.Date.start_of(today, "month")
```
Relays on odoo built-in tooling